src/parser/expression_parser.cpp

```diff
@@ -124,6 +124,18 @@ auto build_condition(auto operator_name, auto &params, auto &data_ids_to_type, a
     return std::make_unique<T>(std::move(matcher), data_id, std::move(arguments), limits);
 }
 
+template <typename T>
+auto build_structured_condition(auto operator_name, auto version, auto &params, auto source,
+    auto &transformers, auto &addresses, auto &limits)
```

Comment:
this would be easier to read if it wasn't all `auto`s and one nondescriptive `T`. Also, shouldn't this be called something like `build_versioned_condition`? what's "structured" here?

Reply:
That's fair, although I refer to multivariate conditions as structured interchangeably.